server/metadata_queue.go

```diff
@@ -0,0 +1,23 @@
+package server
+
+import (
+	"context"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/livepeer-data/pkg/event"
+)
+
+const queuePublishTimeout = 1 * time.Second
```

Comment:
What do you think about accepting this as an arg for the producer and then defining the timeout in `livepeer.go`? That way the producer has a configurable publish timeout and we can just hardcode a timeout we use specifically for go-livepeer in `livepeer.go`.

Reply:
Actually, the producer already has a configurable timeout which can even be sent differently on every publish, which is the `context.Context` argument. This function/timeout definition here was mostly some sugar to avoid needing a go-routine+new ctx with timeout+log embedded in the business logic code. 

So I'm pro making this timeout a configurable flag, will do! But I think we don't need to move it to the producer logic since it is already "more granularly"-configurable for that. Regarding the `BackgroundPublish` sugar, I'm actually considering if I should move it to the `event` lib, or maybe just get rid of it, and do that boilerplate logic inline on the code that sends the event. I'll do one of those, but feel free to lmk any hard preference you might have.